Zend/zend_object_handlers.c

```diff
@@ -680,6 +761,73 @@ ZEND_API zval *zend_std_read_property(zend_object *zobj, zend_string *name, int
 				goto exit;
 			}
 		}
+	} else if (IS_HOOKED_PROPERTY_OFFSET(property_offset)) {
+		zend_function *get = prop_info->hooks[ZEND_PROPERTY_HOOK_GET];
+		if (!get) {
+			if (prop_info->flags & ZEND_ACC_VIRTUAL) {
+				zend_throw_error(NULL, "Property %s::$%s is write-only",
+					ZSTR_VAL(zobj->ce->name), ZSTR_VAL(name));
+				return &EG(uninitialized_zval);
+			}
+			/* Cache the fact that this hook has trivial read. This only applies to
+			 * BP_VAR_R and BP_VAR_IS fetches. */
+			ZEND_SET_PROPERTY_HOOK_SIMPLE_READ(cache_slot);
+
+			retval = OBJ_PROP(zobj, prop_info->offset);
+			if (UNEXPECTED(Z_TYPE_P(retval) == IS_UNDEF)) {
+				/* As hooked properties can't be unset, the only way to end up with an undef
+				 * value is via an uninitialized property. */
+				ZEND_ASSERT(Z_PROP_FLAG_P(retval) == IS_PROP_UNINIT);
+				goto uninit_error;
+			}
+
+			if (UNEXPECTED(type == BP_VAR_W || type == BP_VAR_RW || type == BP_VAR_UNSET)) {
+				if (UNEXPECTED(Z_TYPE_P(retval) != IS_OBJECT)) {
+					zend_throw_error(NULL, "Cannot acquire reference to hooked property %s::$%s",
+						ZSTR_VAL(zobj->ce->name), ZSTR_VAL(name));
+					goto exit;
+				}
+				ZVAL_COPY(rv, retval);
+				retval = rv;
+			}
+			goto exit;
+		}
+
+		guard = zend_get_property_guard(zobj, name);
+		if (UNEXPECTED((*guard) & IN_HOOK)) {
+			if (prop_info->flags & ZEND_ACC_VIRTUAL) {
+				zend_throw_no_prop_backing_value_access(zobj->ce->name, name, /* is_read */ true);
+				return &EG(uninitialized_zval);
+			} else if (UNEXPECTED(!is_in_hook(prop_info))) {
+				zend_throw_forbidden_prop_backing_value_access(zobj->ce->name, name);
```

Comment:
Is a `return` missing here?

Reply:
Indeed, all of these cases require termination.